This code is not compatible with <sys-devel/gcc-4.6.0 wrt http://bugs.gentoo.org/428398
Revert http://cgit.freedesktop.org/udisks/commit/udisks/udisksclient.c?id=b519b97a19060027cd8e1fc1e663dbc61f21f4e4

--- udisks/udisksclient.c
+++ udisks/udisksclient.c
@@ -2215,15 +2215,11 @@
           else if ((g_strcmp0 (id_type[n].version, version) == 0 && strlen (version) > 0) ||
                    (g_strcmp0 (id_type[n].version, "*") == 0 && strlen (version) > 0))
             {
-              /* we know better than the compiler here */
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wformat-nonliteral"
               if (long_string)
                 ret = g_strdup_printf (g_dpgettext2 (GETTEXT_PACKAGE, "fs-type", id_type[n].long_name), version);
               else
                 ret = g_strdup_printf (g_dpgettext2 (GETTEXT_PACKAGE, "fs-type", id_type[n].short_name), version);
               goto out;
-#pragma GCC diagnostic pop
             }
         }
     }
